test/functional/feature_fee_estimation.py

```diff
@@ -383,6 +401,85 @@ def test_acceptstalefeeestimates_option(self):
         assert_equal(self.nodes[0].estimatesmartfee(1)["feerate"], fee_rate)
 
 
+    def sanity_check_cpfp_estimates(self, utxos):
+        """The BlockPolicyEstimator currently does not take CPFP into account. This test
+        sanity checks its behaviour when receiving transactions that were confirmed because
+        of their child's feerate.
+        """
+        # The broadcaster and block producer
+        broadcaster = self.nodes[0]
+        miner = self.nodes[1]
+        # In sat/vb
+        low_feerate = Decimal(2)
+        med_feerate = Decimal(15)
+        high_feerate = Decimal(20)
+
+        self.log.info("Test that fee estimator will ignore all transaction with in block child")
+        # If a transaction got mined and has a child in the same block it was mined
+        # it does not get accounted in the fee estimator.
+        tx = send_tx(wallet=self.wallet, node=broadcaster, utxo=None, feerate=low_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(wallet=self.wallet, node=broadcaster, utxo=u, feerate=high_feerate)
+        self.sync_mempools(wait=0.1, nodes=[broadcaster, miner])
+        self.generate(miner, 1)
+        assert_equal(broadcaster.estimaterawfee(1)["short"]["fail"]["totalconfirmed"], 0)
+
+        # If it has descendants which have a lower ancestor score, it also does not.
+        tx = send_tx(wallet=self.wallet, node=broadcaster, utxo=None, feerate=high_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(wallet=self.wallet, node=broadcaster, utxo=u, feerate=low_feerate)
+        self.sync_mempools(wait=0.1, nodes=[broadcaster, miner])
+        self.generate(miner, 1)
+        assert_equal(broadcaster.estimaterawfee(1)["short"]["fail"]["totalconfirmed"], 0)
+
+        # Even if it's equal fee rate.
+        tx = send_tx(wallet=self.wallet, node=broadcaster, utxo=None, feerate=high_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(wallet=self.wallet, node=broadcaster, utxo=u, feerate=high_feerate)
+        self.sync_mempools(wait=0.1, nodes=[broadcaster, miner])
+        self.generate(miner, 1)
+        assert_equal(broadcaster.estimaterawfee(1)["short"]["fail"]["totalconfirmed"], 0)
```

Comment:
This is an easy candidate to remove code duplication by extracting this out in an internal function, and by accepting `tx` and `feeRate` as function args.

Reply:
Yes, this is fixed